.github/workflows/auto-security-secrets.yml

```
@@ -0,0 +1,49 @@
+      - name: Run Gitleaks
+        uses: gitleaks/gitleaks-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITLEAKS_LICENSE: ${{ secrets.GITLEAKS_LICENSE }} # Optional: for enterprise features
+
+      - name: Upload SARIF report
+        if: failure()
+        uses: github/codeql-action/upload-sarif@v3
+        with:
+          sarif_file: results.sarif
+          category: secret-scanning
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)